### PR TITLE
[DDW-39] Acceptance tests for wallets, transactions and addresses ordering

### PR DESCRIPTION
**Pull request by daniloprates**
This PR creates acceptance tests for wallets, transactions and addresses ordering.

## Todo:

- [x] acceptance tests for wallets ordering;
- [x] acceptance tests for transactions ordering;
- [x] acceptance tests for addresses ordering.

---

### Basics

- [x] PR is updated to the most recent version of target branch (and there are no conflicts)
- [x] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [x] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`npm run test`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`npm run dev`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`npm run package` / CI builds)
- [x] There are no *flow* errors or warnings (`npm run flow:test`)
- [x] There are no *lint* errors or warnings (`npm run lint`)
- [x] Text changes are proofread and approved (Jane Wild)
- [x] There are no missing translations (running `npm run manage:translations` produces no changes)
- [x] UI changes look good in all themes (Alexander Rukin)
- [x] Storybook works and no stories are broken (`npm run storybook`)
- [x] In case of npm dependency changes both `package-lock.json` and `yarn.lock` files are updated

### Code Quality
- [x] Important parts of the code are properly documented and commented
- [x] Code is properly typed with flow
- [x] React components are split-up enough to avoid unnecessary re-rendering
- [x] Any code that only works in Electron is neatly separated from components

### Testing
- [x] New feature / change is covered by acceptance tests
- [x] All existing acceptance tests are still up-to-date
- [x] New feature / change is covered by Daedalus Testing scenario
- [x] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: features/step_definitions/transactions-steps.js [hunk 1]
```diff
@@ -72,3 +72,27 @@ Then(/^I should see the following transactions:$/, async function (table) {
 When(/^I click on the show more transactions button$/, async function () {
   await this.waitAndClick('.WalletTransactionsList_showMoreTransactionsButton');
 });
+
+Then(/^The transactions should be ordered from newest to oldest$/, async function () {
```
Comment (nikolaglumac) [inline features/step_definitions/transactions-steps.js]: @daniloprates please remove this test step (it is a duplicate)

File: features/step_definitions/transactions-steps.js [hunk 2]
```diff
@@ -72,3 +72,27 @@ Then(/^I should see the following transactions:$/, async function (table) {
...
+
+  const [firstTransaction, lastTransaction] = await this.client.getText('.Transaction_amount');
+
+  await this.client.waitUntil(async () =>
+    expect(firstTransaction).to.equal('2.000000') && expect(lastTransaction).to.equal('1.000000')
+  );
+
+});
+
+Then(/^The transactions should be grouped by their date$/, async function () {
```
Comment (nikolaglumac) [inline features/step_definitions/transactions-steps.js]: @daniloprates please remove this test step (it is a duplicate)

File: features/step_definitions/receive-steps.js [hunk 3]
```diff
@@ -16,3 +16,15 @@ Then('I should see {int} addresses', async function (numberOfAddresses) {
   const addressesFound = await getVisibleElementsCountForSelector(this.client, '.WalletReceive_walletAddress', `.generatedAddress-${numberOfAddresses}`);
   expect(addressesFound).to.equal(numberOfAddresses);
 });
+
+Then('They should be ordered by created date descending', async function () {
```
Comment (nikolaglumac) [inline features/step_definitions/receive-steps.js]: @daniloprates please update this test step to take the input values in a table form.

File: features/step_definitions/wallets-steps.js [hunk 4]
```diff
@@ -430,3 +430,15 @@ Then(/^I should see newly generated address as active address on the wallet rece
     return generatedAddress === activeAddress;
   });
 });
+
+Then(/^the wallets should be ordered from oldest to newest$/, async function () {
+
+  const [wallet1, wallet2, wallet3] = await this.client.getText('.SidebarWalletMenuItem_title');
```
Comment (nikolaglumac) [inline features/step_definitions/wallets-steps.js]: @daniloprates please update this test step to take the input values in a table form.

File: features/step_definitions/receive-steps.js [hunk 5]
```diff
@@ -16,3 +16,22 @@ Then('I should see {int} addresses', async function (numberOfAddresses) {
...
   expect(addressesFound).to.equal(numberOfAddresses);
 });
+
+Then('I should see the following addresses:', async function (table) {
+  const expectedAdresses = table.hashes();
+  let addresses;
+
+  await this.client.waitUntil(async () => {
+    addresses = await this.client.getAttribute('.WalletReceive_walletAddress','class');
+    return addresses.length === 3;
```
Comment (DominikGuzei) [inline features/step_definitions/receive-steps.js]: This hardcoded number is not good 😉

Comment (DominikGuzei) [reply features/step_definitions/receive-steps.js]: And i would write the whole expectation differently … more like this:

Comment (daniloprates) [reply features/step_definitions/receive-steps.js]: @DominikGuzei thanks. very well spotted!

**Comment by nikolaglumac** [comment]
@DominikGuzei @daniloprates you have a green light to merge this one from my side. I am unsure if there are some more changes @DominikGuzei want's to add?

**Pull request merged**